core/java8/proxy/src/main/java/org/apache/openwhisk/runtime/java/action/Proxy.java

```diff
@@ -48,7 +51,20 @@ public Proxy(int port) throws IOException {
 
         this.server.createContext("/init", new InitHandler());
         this.server.createContext("/run", new RunHandler());
-        this.server.setExecutor(null); // creates a default executor
+
+        if (Boolean.parseBoolean(System.getenv("__OW_ALLOW_CONCURRENT"))) {
+            ThreadPoolExecutor executor = new ThreadPoolExecutor(
+                    10,                  // Core size.
```

Comment:
I'm not sure how we should optimally determine this number since every function will have a different concurrency setting. Should it err on the side of maximizing low concurrency, i.e. most functions probably won't increase their setting past 1.

Reply:
Agreed.